lib/cli-engine/config-array-factory.js

```diff
@@ -500,6 +500,14 @@ class ConfigArrayFactory {
              */
             element.criteria = OverrideTester.and(criteria, element.criteria);
 
+            /*
+             * Remove `root` property to ignore `root` settings which came from
+             * `extends` in `overrides`.
+             */
+            if (element.criteria) {
+                element.root = void 0;
```

Comment:
I don't have a strong preference here, but would it maybe be better to use `delete element.root`?

Reply:
Actually, I hate `delete` expressions because it kills the V8's runtime optimization; if we use `delete` to an object, V8 no longer handles the object as having a stable structure. This is fine since every element has the same structure ([lib/cli-engine/config-array-factory.js#L550-L568](https://github.com/eslint/eslint/blob/ec105b24f7e036ecdc4267f018529ac3765e29d5/lib/cli-engine/config-array-factory.js#L550-L568)).

Reply:
good point!

Reply:
Thanks for the explanation. I'll mark this as resolved.